File: Algorithms/src/812. Largest Triangle Area/Code/solution.py
from typing import List

class Solution:
    def largestTriangleArea(self, points: List[List[int]]) -> float:
        n = len(points)
        max_area = 0.0
        # iterate over all combinations of three distinct points
        for i in range(n - 2):
            for j in range(i + 1, n - 1):
                for k in range(j + 1, n):
                    x1, y1 = points[i]
                    x2, y2 = points[j]
                    x3, y3 = points[k]
                    # shoelace/cross product formula gives doubled area
                    doubled = abs(x1*(y2 - y3) + x2*(y3 - y1) + x3*(y1 - y2))
                    area = doubled * 0.5
                    if area > max_area:
                        max_area = area
        return max_area

File: Algorithms/src/812. Largest Triangle Area/Code/solution.cpp
class Solution {
public:
    double largestTriangleArea(vector<vector<int>>& points) {
        int n = points.size();
        double maxArea = 0.0;
        // iterate over all triples i < j < k
        for (int i = 0; i < n-2; ++i) {
            for (int j = i+1; j < n-1; ++j) {
                for (int k = j+1; k < n; ++k) {
                    // coordinates
                    int x1 = points[i][0], y1 = points[i][1];
                    int x2 = points[j][0], y2 = points[j][1];
                    int x3 = points[k][0], y3 = points[k][1];
                    // shoelace / cross-product for double area
                    double doubled = fabs( (double)x1*(y2 - y3)
                                         + (double)x2*(y3 - y1)
                                         + (double)x3*(y1 - y2) );
                    double area = doubled * 0.5;
                    if (area > maxArea) maxArea = area;
                }
            }
        }
        return maxArea;
    }
};
